### PR TITLE
Query workspaces rules

### PR DESCRIPTION
**Pull request by hariseldon78**
#### Describe your PR, what does it fix/add?
It adds a command to query the defined workspace rules.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
This will be useful for automating behaviors, for example in the Waybar (like mentioned in https://github.com/Alexays/Waybar/issues/2271).

#### Is it ready for merging, or does it need work?
Tested and ready. 

Here is an example: for this configuration:
```
workspace=name:testpers,persistent:true
workspace=name:testnonpers,persistent:false
workspace=name:testnormal
workspace=name:testanother
workspace=name:myworkspace,gapsin:0,gapsout:0
workspace=DP-1,3,rounding:false,bordersize:0
```

These are the json and plain outputs:
```
> hyprctl workspacerules -j
[{
    "workspaceString": "name:testpers",
    "persistent": true
},{
    "workspaceString": "name:testnonpers"
},{
    "workspaceString": "name:testnormal"
},{
    "workspaceString": "name:testanother"
},{
    "workspaceString": "name:myworkspace",
    "gapsIn": 0,
    "gapsOut": 0
},{
    "workspaceString": "3",
    "monitor": "DP-1",
    "default": true,
    "borderSize": 0,
    "rounding": false
}]

> hyprctl workspacerules
workspace=name:testpers,persistent:true
workspace=name:testnonpers
workspace=name:testnormal
workspace=name:testanother
workspace=name:myworkspace,gapsIn:0,gapsOut:0
workspace=3,monitor:DP-1,default:true,borderSize:0,rounding:false
```

### CODE REVIEW

File: src/debug/HyprCtl.cpp [hunk 1]
```diff
@@ -229,6 +229,44 @@ static std::string getWorkspaceData(CWorkspace* w, HyprCtl::eHyprCtlOutputFormat
...
+        const std::string shadow     = (bool)(r.shadow) ? std::format(",\n    \"shadow\": {}", boolToString(r.shadow.value())) : "";
+
+        std::string       result = std::format(R"#({{
+    "workspaceString": "{}"{}{}{}{}{}{}{}{}
+}})#",
+                                               escapeJSONStrings(r.workspaceString), monitor, default_, persistent, gapsIn, gapsOut, borderSize, border, rounding, decorate, shadow);
+
+        return result;
+    } else {
+        const std::string monitor    = r.monitor.empty() ? "" : std::format(",monitor:{}", escapeJSONStrings(r.monitor));
```
Comment (vaxerski) [inline src/debug/HyprCtl.cpp:253]: I don't like this syntax. Standard hyprctl syntax would be:
```
Workspace rule <name>:
\t<prop>: <value>
```

Unset props probably should still be printed here as `unset`.

Comment (hariseldon78) [reply src/debug/HyprCtl.cpp:253]: I was trying to emulate the configuration string but ok, no problem.

Comment (hariseldon78) [reply src/debug/HyprCtl.cpp]: here it is: 
Workspace rule 3:
	monitor:DP-1
	default:true
	persistent:<unset>
	gapsIn:<unset>
	gapsOut:<unset>
	borderSize:0
	border:<unset>
	rounding:false
	decorate:<unset>
	shadow:<unset>

Comment (vaxerski) [reply src/debug/HyprCtl.cpp]: missing spaces before args: `border: <unset>`

other than that, lgtm

**Review by vaxerski — APPROVED** [review]
lgtm, thanks!

**Pull request merged**